WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductCard.swift

```diff
@@ -1,7 +1,87 @@
 import Yosemite
 import SwiftUI
 
-struct CollapsibleProductRowCard: View {
+/// Displays a single collapsible product row or grouped parent and child product rows
+struct CollapsibleProductCard: View {
+    @ObservedObject var viewModel: ProductRowViewModel
+
+    /// Used for tracking order form events.
+    let flow: WooAnalyticsEvent.Orders.Flow
+
+    /// Handles when a discount is added to the row with the provided `ProductRowViewModel` ID
+    var onAddDiscount: (Int64) -> Void
+
+    /// Tracks if discount editing should be enabled or disabled. False by default
+    var shouldDisableDiscountEditing: Bool = false
+
+    /// Tracks if discount should be allowed or disallowed.
+    var shouldDisallowDiscounts: Bool = false
```

Comment:
super nit: I think these can all be private let
```suggestion
    private let onAddDiscount: (Int64) -> Void

    /// Tracks if discount editing should be enabled or disabled. False by default
    private let shouldDisableDiscountEditing: Bool

    /// Tracks if discount should be allowed or disallowed.
    private let shouldDisallowDiscounts: Bool
```

Reply:
Both `shouldDisableDiscountEditing` and `shouldDisallowDiscounts` need to be `var` because they are mutable (mutated in `EditableOrderViewModel`). The first depends on whether the order is syncing remotely, and the second depends on whether the order has coupons.

But `onAddDiscount` and `flow` can both be `private let`, good call; I didn't think to change them from how they were done originally in `CollapsibleProductRowCard`.